src/Avalonia.Controls/SplitView.cs

```diff
@@ -497,5 +500,45 @@ private void PaneChanged(AvaloniaPropertyChangedEventArgs e)
                 LogicalChildren.Add(newChild);
             }
         }
+
+        /// <summary>
+        /// Called when the <see cref="IsPaneOpen"/> property has to be coerced.
+        /// </summary>
+        /// <param name="value">The value to coerce.</param>
+        protected virtual bool OnCoerceIsPaneOpen(bool value)
+        {
+            if (value)
+            {
+                OnPaneOpening(EventArgs.Empty);
+            }
+            else
+            {
+                var eventArgs = new SplitViewPaneClosingEventArgs(false);
+                OnPaneClosing(eventArgs);
+
+                if (eventArgs.Cancel)
+                {
+                    return !value;
+                }
```

Comment:
I'm not using the Expander's hack to raise a property changed notification here when the value is reset. I don't think it is needed because we don't have to worry about a desync with a ToggleButton in the control template like with Expander.

If this assumption is wrong, hopefully it can be fixed correctly with a property system fix (https://github.com/AvaloniaUI/Avalonia/issues/10092). The code shouldn't need to change here in the future.